packages/get-context/src/index.ts

```diff
@@ -119,10 +123,10 @@ export default async function getContext<T> (
     })
   })
   if (opts.hooks?.readPackage) {
-    projects = projects.map((project) => ({
-      ...project,
-      manifest: opts.hooks!.readPackage!(project.manifest),
-    }))
+    for (const project of importersContext.projects) {
+      project.unhookedManifest = project.manifest
+      project.manifest = opts.hooks!.readPackage!(JSON.parse(JSON.stringify(project.manifest)))
```

Comment:
Hmm... not a good idea to clone an object this way. From StackOverflow:

```js
const a = {
  string: 'string',
  number: 123,
  bool: false,
  nul: null,
  date: new Date(),  // stringified
  undef: undefined,  // lost
  inf: Infinity,  // forced to 'null'
  re: /.*/,  // lost
}
console.log(a);
console.log(typeof a.date);  // Date object
const clone = JSON.parse(JSON.stringify(a));
console.log(clone);
console.log(typeof clone.date);  // result of .toISOString()
```

Please use something like `lodash.clonedeep` instead.

Reply:
We have ramda, so `R.clone` may be used.

But I don't think in this case it is an issue because the manifest is the raw package.json file's content, which will not have these issues

Reply:
`R.clone` is probably more performant than a round trip through JSON.

Reply:
maybe. OK. Let's change it

Reply:
I'll leave that up to you.